loqusdb/plugins/mongo/structural_variant.py

```diff
@@ -134,14 +117,103 @@ def add_structural_variant(self, variant, max_window = 3000):
 
+    def delete_structural_variant(self, variant, max_window = 3000):
+
+        # This will return the cluster most similar to variant or None
+        cluster = self.get_structural_variant(variant)
+
+        if cluster is None:
+            return
+
+        case_id = variant.get('case_id')
+
+        if case_id in cluster['families']:
+            # Remove case from the array
+            cluster['families'].remove(case_id)
+
+        nr_events = cluster['observations'] - 1
+
+        if nr_events == 0:
+            # Remove SV if no observations remain
+            self.db.structural_variant.delete_one({'_id': cluster['_id']})
+
+            # remove identity
+            identity_obj = Identity(cluster_id=cluster['_id'], variant_id=variant['id_column'],
+                                    case_id=case_id)
+            self.db.identity.delete_one(dict(identity_obj))
+
+            return
+
+        pos_mean = int((cluster['pos_sum'] - variant['pos']) // (nr_events))
+        end_mean = int((cluster['end_sum'] - variant['end']) // (nr_events))
+
+        cluster_len, interval_size = self._update_sv_metrics(sv_type=cluster['sv_type'],
+                                                             pos_mean=pos_mean,
+                                                             end_mean=end_mean,
+                                                             max_window=max_window)
+
+        res = self.db.structural_variant.find_one_and_update(
+            {'_id': cluster['_id']},
+            {
+                '$inc': {
+                    'observations': -1,
+                    'pos_sum': -variant['pos'],
+                    'end_sum': -variant['end'],
+                },
+
+                '$set': {
+                    'pos_left': max(pos_mean - interval_size, 0),
+                    'pos_right': pos_mean + interval_size,
+                    'end_left': max(end_mean - interval_size, 0),
+                    'end_right': end_mean + interval_size,
+                    'families': cluster['families'],
+                    'length': cluster_len,
+                }
+            }
+        )
+
+        # Insert an identity object to link cases to variants and clusters
+        identity_obj = Identity(cluster_id=cluster['_id'], variant_id=variant['id_column'],
+                                case_id=case_id)
+        self.db.identity.delete_one(dict(identity_obj))
+
+        return
+
+    def _update_sv_metrics(self, sv_type, pos_mean, end_mean, max_window):
+        # We need to calculate the new cluster length
+        # Handle translocation as a special case
+        if sv_type != 'BND':
+            cluster_len = end_mean - pos_mean
+            # We need to adapt the interval size depending on the size of the cluster
+            divider = 10
+            if cluster_len < 1000:
+                # We allow intervals for smaller variants to be relatively larger
+                divider = 2
+            elif cluster_len < 1000:
```

Comment:
Any reason why this elif condition is the same as the above if condition? @moonso

Reply:
```suggestion
            elif cluster_len < 10000:
```
It should be 10000